### DISCUSSION TITLE
gatsby-source-shopify does not show allShopifyArticle to query with graphql with no blogs

### DISCUSSION THREAD

**Discussion by jordanpsleeper** [gatsbyjs/gatsby#29777] [Help]
Working with a gatsby site that feeds multiple shopify stores and using this plugin, if the shopify store doesn't have any active or visible articles the graphql allShopifyArticle is not available to query. 

This throws error on stores that haven't used their blogs yet. Is this the intended behavior or is there a work around for testing if allShopifyArticle exists before trying to use it?

**Comment by SabbeRubbish** [comment]
**Update:** new insight in next post.

Hi jordanpsleeper,

Part of working with GraphQL is of course its dynamic nature.
But that dynamic and automatic behavior has a downside.

I've found quite some articles complaining about this. What you could do is launch an introspection query into GraphQL to find out whether the type exists:
```
{
  __type(name: "ShopifyArticle") {
    name
  }
}
```
Would result in:
```
{
  "data": {
    "__type": {
      "name": "ShopifyArticle"
    }
  },
  "extensions": {}
}
```

However the ``name`` property will be ``null`` if the type doesn't exist.
I'm not sure yet how you could prevent your page query not to launch, but you could conditionally let your StaticQuery run based on the existence of the type.

Hope this helps you in the right direction.

**Reply by SabbeRubbish** [reply]
On second thought, maybe consider actually defining the type explicitly in your `gatsby-node.js`.
It's not required, but it may fix your node missing. Also, you could propose this to the Spotify source plugin team to always provide the GraphQL type.

Basically you just need to implement the createSchemaCustomization API and add the type:
```
// gatsby-node.js
exports.createSchemaCustomization = ({ actions, schema }) => {
  const { createTypes } = actions
  createTypes(`
      type ShopifyArticle implements Node {
        id: String!
      }
    `)
}
```
Thanks to inferences, all your other fields will populate as well.
I've tested this in a simple setup with no plugins, and the allShopifyArticle node appears.

Not sure if this conflicts with the type that Spotify makes.
Hope this helps you.
